pkg/jx/cmd/step/create/step_create_task.go

```diff
@@ -236,7 +237,7 @@ func (o *StepCreateTaskOptions) Run() error {
 				}
 			}()
 		}
-		err := o.mergePullRefs(o.cloneDir)
+		pr, err = o.mergePullRefs(o.cloneDir)
```

Comment:
The thing which keeps confusing me here is that it seems that `PULL_REFS` is only passed via an env variable. Almost everything else we pass via option. This seems a bit arbitrary to me. I think we need a better plan here in general. 

One thing we could arleady do now is to have some validateOptions at the beginning of `Run` which determines what type of build we are dealing with, validates we have all parameters, parses the pull refs if needed.

Reply:
They are actually passed as `o.CustomEnvs` and it's probably to keep the spec of Prow: https://github.com/oracle/kubernetes-test-infra/tree/master/prow (at Job Environment Variables).
This way we can always "check" we are respecting the spec.

Reply:
> They are actually passed as o.CustomEnvs and it's probably to keep the spec of Prow

Sure. But why do our controllers not also only act on env variables? Instead, there is this mixture of options and env variables which I find confusing.